tests/test-utils.js

```diff
@@ -0,0 +1,178 @@
+                resolve(result.toString());
+            })
+        );
+    });
+};
+
+const _throwFatalError = (message) => {
+    throw new Error(`${red(ERROR_OUTPUT_STRING)} ${message}`);
+};
+
+// #endregion Private Functions
+
+// -----------------------------------------------------------------------------------------
+// #region Public Functions
+// -----------------------------------------------------------------------------------------
+
+/**
+ * The setup of this Node spawn wrapper is heavily based on this article:
+ * https://medium.com/@zorrodg/integration-tests-on-node-js-cli-part-1-why-and-how-fa5b1ba552fe
+ *
+ * It allows for us to call synchronous or asynchronous parent-level commands and test the layer
+ * that glues our modules together.
+ */
+const testUtils = {
+    /**
+     * Creates and enters a temporary working directory. Returns the directory name as well as
+     * a cleanup function for resetting the working directory and removing created dir/files.
+     *
+     * @param {string} [prefix="tmp"] (Optional) Prefix for the directory name to be generated.
+     */
+    createAndUseTmpDir(prefix = "tmp") {
+        const oldPwd = shell.pwd().toString();
+        const tmpDir = upath.toUnix(path.join(oldPwd, `${prefix}-${faker.random.uuid()}`));
+
+        shell.mkdir("-p", tmpDir);
+        shell.cd(tmpDir);
+
+        const cleanupTmpDir = () => {
+            shell.cd(oldPwd);
+            shell.rm("-rf", tmpDir);
+        };
+
+        return { cleanupTmpDir, tmpDir };
+    },
+    /**
+     * Wrapper around `child_process.spawn` that passes in the main executable, `cli.js`, as the process path.
+     * This allows you to pass just the command you're testing in as the first argument.
+     *
+     * @example
+     * const result = await testUtils.executeCliCommand("dotnet", ["--help"]);
+     * @param {string} command
+     * @param {string[]} args
+     * @param {object} opts
+     */
+    executeCliCommand(command, args = [], opts = {}) {
+        // Generate the absolute path of the main executable file (cli.js) based on the current
+        // file's directory.
+        const cliEntrypointPath = upath.toUnix(path.join(__dirname, "..", "cli.js"));
+        return _executeNode(cliEntrypointPath, [command, ...args], opts);
+    },
+    /**
+     * Calls `child_process.spawnSync()` with the given args or options. If the process fails,
+     * it will throw a fatal error to prevent further execution. Useful for arrangement steps
+     * that require setting up data for the system under test.
+     *
+     * @param {string} processPath
+     * @param {string[]} [args=[]]
+     * @param {object} [opts={}]
+     */
+    executeOrThrow(processPath, args = [], opts = {}) {
+        const { status } = child_process.spawnSync(processPath, args, opts);
+        if (status !== 0) {
+            _throwFatalError(`'${processPath} ${args.join(" ")}' failed with exit code: ${status}`);
+        }
+    },
+    verifyDotnetPath() {
```

Comment:
Feels like it should just be a 'verify' function in the dotnet-path module.

Reply:
Good call - moved it to `dotnetPath` and added unit tests.